src/core/vdom/create-component.js

```diff
@@ -287,6 +287,17 @@ function extractProps (data: VNodeData, Ctor: Class<Component>): ?Object {
   if (attrs || props || domProps) {
     for (const key in propOptions) {
       const altKey = hyphenate(key)
+      const keyInLowerCase = key.toLowerCase()
```

Comment:
Can we move this inside the `NODE_ENV` conditional block as well? It's ok to use two nested ifs.